### ISSUE TITLE
If you log out & in without setting up recovery, you get an unactionable "Confirm your identity" dialog

### ISSUE THREAD

**Issue by ara4n** [element-hq/element-web#27015]
[Updated 2026-02-16]
 
1. Sign up
2. Immediately log out, before setting up 4S
3. Log back in:
<img width="2147" height="1496" alt="Image" src="https://github.com/user-attachments/assets/e6086f6c-f63d-4bbf-bc71-f70f1e325cf4" />

There is no clear action for the user to take here. The "learn more" button takes you to an unhelpful part of our FAQ. I want to confirm, and don't want to click "Can't confirm?" yet.

Because I don't have any other devices and haven't set up a recovery key, I do need to reset my identity, but I was paralysed at this stage.

It would be much clearer if the user was simply told that they must reset their crypto identity (taken straight to a "You must reset your cryptographic identity page) and how to avoid this in the future.

**Comment by BillCarsonFr** [comment]
Maybe we can lower a bit the tone of the copy.
But you will have to reset the cross signing keys, this will require re-authentication

**Comment by mxandreas** [comment]
> On logout: Explain that it will happen if you don't set up recovery?

Yes, this is the solution already on EX, and it is also planned to be added on EW as part of the invisible crypto rollout. See https://github.com/element-hq/element-meta/issues/2834.

**Comment by anoadragon453** [comment]
I tried signing up on the chat.fosdem.org (non-MAS) homeserver instance recently and ran into this issue. I didn't set up a recovery key as I was not forced to on sign-up, and didn't join any rooms/send any messages, so I didn't receive a toast to set up a recovery key. I also saw another user during FOSDEM just ignore this toast.

After signing up, logging in and logging out of one device, my experience when logging in on a new device was confusing. This is the screen I received after verifying my username/password:

<img width="2147" height="1496" alt="Image" src="https://github.com/user-attachments/assets/e6086f6c-f63d-4bbf-bc71-f70f1e325cf4" />

There is no clear action for the user to take here. The "learn more" button takes you to an unhelpful part of our FAQ. I *want* to confirm, and don't want to click "Can't confirm?" yet.

Because I don't have any other devices and haven't set up a recovery key, I do *need* to reset my identity, but I was paralysed at this stage.

It would be much clearer if the user was simply told that they must reset their crypto identity (taken straight to a "You must reset your cryptographic identity page) and how to avoid this in the future.

---

FYI if you do have another device, this is the screen you get, which is much clearer:

<img width="2776" height="1666" alt="Image" src="https://github.com/user-attachments/assets/d9d0c4df-6757-4866-9b2d-75c7bf51797c" />

**Comment by mxandreas** [comment]
> It would be much clearer if the user was simply told that they must reset their crypto identity (taken straight to a "You must reset your cryptographic identity page) and how to avoid this in the future.

The point on confusion is fair. The assumption that it would be much clearer if if we told immediately that "they must reset their crypto identity" - may or may not be true. Many users have no glue what a reset is and they just follow the buttons, anyway. And regardless how clear this UI would be, it is still a "game over" UI which is always perceived as negative and annoying - even if you actually did not lose much history.

Hence, it is probably more important to make sure users do set up recovery https://github.com/element-hq/element-meta/issues/3025 and that we are more clear about the consequences before they log out https://github.com/element-hq/element-meta/issues/2834

**Comment by anoadragon453** [comment]
> Hence, it is probably more important to make sure users do set up recovery [element-hq/element-meta#3025](https://github.com/element-hq/element-meta/issues/3025) and that we are more clear about the consequences before they log out [element-hq/element-meta#2834](https://github.com/element-hq/element-meta/issues/2834)

Presumably we can do both?

**Comment by parasiteoflife** [comment]
Would it be possible to also not show this "verify identity" when the server has encryption disabled?

**Comment by richvdh** [comment]
> Would it be possible to also not show this "verify identity" when the server has encryption disabled?

What exactly do you mean by "the server has encryption disabled"?

**Comment by richvdh** [comment]
I've updated the description to reflect the current state of this issue.
